src/latex/generator.ml

```diff
@@ -200,15 +209,18 @@ and href ppf (l,txt) =
 and large_table size ppf tbl =
     let rec row ppf = function
       | [] -> Raw.break ppf Line
-      | [a] -> pp ppf a
+      | [a] -> pp ppf a; Raw.break ppf Line
+      | [a;b] ->
+        Fmt.pf ppf "%a%a%a"
+          pp a
+          Raw.break Aesthetic
+          (Raw.indent pp) b
       | a :: (_ :: _ as q) ->
         Fmt.pf ppf "%a%a%a"
           pp a
           Raw.break Aesthetic
           (Raw.indent row) q  in
-    let matrix ppf m =
-
-      List.iter (row ppf) m in
+    let matrix ppf m = List.iter (row ppf) m in
     match size with
     | Huge -> Raw.break ppf Line; matrix ppf tbl
     | Large | _ -> Raw.indent matrix ppf tbl
```

Comment:
In the `Huge` case, the output now always begins and ends with a Line break. In the `Large|Small|Empty` case, your change to the output guarantees that we always have a Line break at the end, but not at the beginning. On the other hand, small-height small tables get a Line break at both the beginning and end. This sounds a bit inconsistent, but I don't know what the intent is.

Reply:
The Huge case is in fact unnecessary, this was probably some scoria from prior implementation of recursive tables.

Reply:
With the new implementation I'm still confused by line breaks in the code: if I understand correctly, `row` produces a final line break on lists of size 0 or 1, but not 2 and above, and it's not clear to me why.

Reply:
There is probably too much hardwired knowledge about the imbrication of different environments, but here the indentation environment provides the line break for rows with more than one elements.